File: Data-Structure-I/121-Best-Time-to-Buy-and-Sell-Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_elem = prices[0]
        ans = 0
        for i in range(1, len(prices)):
            min_elem = min(prices[i], min_elem)
            ans = max(ans, prices[i] - min_elem)
        return ans

File: Data-Structure-I/121-Best-Time-to-Buy-and-Sell-Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = INT_MIN;
        int min_elem = INT_MAX;
        for(auto i : prices){
            min_elem = min(i, min_elem);
            ans = max(ans, i - min_elem);
        }
        return ans;
    }
};
